SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -183,9 +193,48 @@ void CFluidFlamelet::PreprocessLookUp(CConfig* config) {
 
   /*--- Passive look-up terms ---*/
   size_t n_lookups = config->GetNLookups();
-  varnames_LookUp.resize(n_lookups);
-  val_vars_LookUp.resize(n_lookups);
-  for (auto iLookup = 0u; iLookup < n_lookups; iLookup++) varnames_LookUp[iLookup] = config->GetLookupName(iLookup);
+  if (n_lookups == 0) {
+    varnames_LookUp.resize(1);
+    val_vars_LookUp.resize(1);
+    varnames_LookUp[0] = "NULL";
+  } else {
+    varnames_LookUp.resize(n_lookups);
+    val_vars_LookUp.resize(n_lookups);
+    for (auto iLookup = 0u; iLookup < n_lookups; iLookup++) varnames_LookUp[iLookup] = config->GetLookupName(iLookup);
+  }
+
+  /*--- Preferential diffusion scalars ---*/
+  varnames_PD.resize(FLAMELET_PREF_DIFF_SCALARS::N_BETA_TERMS);
+  val_vars_PD.resize(FLAMELET_PREF_DIFF_SCALARS::N_BETA_TERMS);
+
+  varnames_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_PROGVAR] = "Beta_ProgVar";
+  varnames_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_ENTH_THERMAL] = "Beta_Enth_Thermal";
+  varnames_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_ENTH] = "Beta_Enth";
+  varnames_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_MIXFRAC] = "Beta_MixFrac";
+
+  val_vars_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_PROGVAR] = beta_progvar;
+  val_vars_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_ENTH_THERMAL] = beta_enth_thermal;
+  val_vars_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_ENTH] = beta_enth;
+  val_vars_PD[FLAMELET_PREF_DIFF_SCALARS::I_BETA_MIXFRAC] = beta_mixfrac;
+
+  PreferentialDiffusion = false;
+  switch (Kind_DataDriven_Method) {
+    case ENUM_DATADRIVEN_METHOD::LUT:
+      PreferentialDiffusion = look_up_table->CheckForVariables(varnames_PD);
+      break;
+    case ENUM_DATADRIVEN_METHOD::MLP:
+#ifdef USE_MLPCPP
+      n_betas = 0;
+      for (auto iMLP = 0u; iMLP < config->GetNDataDriven_Files(); iMLP++) {
+        auto outputMap = lookup_mlp->FindVariableIndices(iMLP, varnames_PD, false);
+        n_betas += outputMap.size();
+      }
+      PreferentialDiffusion = (n_betas == varnames_PD.size());
```

Comment:
It is easier to change a config option than change a lookup table, can you additionally introduce a config option to switch on preferential diffusion?

Reply:
True, but it does add another config option, which I'm trying to limit, since this functionality already has quite some new options. I'll add the option back for now.